tests/test_voting.py

```diff
@@ -235,14 +235,41 @@ def verify_vote_schema(cls, vote):
         'other_previous_block': {},
     }
 
+# TODO: Check all other tests in this class for similar mistakes
+
 
 @patch('bigchaindb.voting.Voting.verify_vote_signature', return_value=True)
 def test_duplicate_vote_throws_critical_error(b):
     keyring = 'abc'
+
+    # block_election calls count_votes that calls `cls.verify_vote_schema`.
     block = {'id': 'xyz', 'block': {'voters': 'ab'}}
+    # As `vote` has a required property `signature`, the vote here should
+    # technically be counted as `malformed` (or invalid), rendering this test
+    # incorrect as `CriticalDuplicateVote` could be trigger elsewhere.
     votes = [{
+        # according to the vote's schema a public key needs to have up to 43
+        # chars, making the keyring in this test to not pass
+        # `verify_vote_schema`.
         'node_pubkey': c,
         'vote': {'is_block_valid': True, 'previous_block': 'a'}
     } for c in 'aabc']
     with pytest.raises(CriticalDuplicateVote):
         Voting.block_election(block, votes, keyring)
+    # TODO: Fix this test
+    assert False
+
+
+@patch('bigchaindb.voting.Voting.verify_vote_signature', return_value=True)
+@patch('bigchaindb.voting.Voting.verify_vote_schema', return_value=True)
```

Comment:
This is not idea. Ideally we setup the test such that we can remove patching this method. @vrde and I just used it to save time.

Reply:
I don't mind patching `verify_vote_schema` since we are testing for a malicious, valid vote. We also have `test_verify_vote_schema` that should cover the schema validation.